### PR TITLE
fix #422, add worker rest api

### LINKED ISSUE #422: [Dashboard] create a UI for master informations
Master task is #11 

The UI should contains information about workers and applications.

The worker information includes:

“id”: string, 
   “state”: string, 
Host:
Port:
Alive for:
Log path:
config
   executors:
       executor1: 
appId, 
executorId
slots taken
       executor2: 
appId, 
executorId
slots taken

   available slots:
   “resources”: [
        {“total slots”:  int}, //…
   ]

and system load information. (like disk space, memory, CPU, load)

The application information includes:

“name”: string,
   “appmaster”: string,
appId: Int
status: running or dead.
submission date:    
“application url”  //for redirect
Username:
Executors:
   Executor1:
       Worker1
AppMaster address: 
    Worker xx

### PR DESCRIPTION
**Pull request by manuzhang**
[Review on Reviewable](https://reviewable.io/reviews/intel-hadoop/gearpump/448)

### CODE REVIEW

File: core/src/main/scala/org/apache/gearpump/cluster/worker/Worker.scala [hunk 1]
```diff
@@ -99,6 +110,13 @@ private[cluster] class Worker(masterProxy : ActorRef) extends Actor with TimeOut
       context.stop(self)
     case UpdateResourceSucceed =>
       LOG.info(s"Worker $id update resource succeed")
+    case GetWorkerData(workerId) =>
+      val aliveFor = System.currentTimeMillis() - createdTime
+      val address = context.system.asInstanceOf[ExtendedActorSystem].provider.getDefaultAddress
+      val host = address.host.get
+      val port = address.port.get
+      sender ! WorkerData(Some(WorkerDescription(id, "active", host, port,
+        aliveFor, logFile, configStr, executorsInfo.values.toArray, resource.slots)))
```
Comment (kkasravi) [inline core/src/main/scala/org/apache/gearpump/cluster/worker/Worker.scala]: Would it be valuable to return process state like {RUNNING, ZOMBIE, ...}?

File: services/src/test/scala/org/apache/gearpump/services/WorkersServiceSpec.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  override def afterEach: Unit = {
+    restUtil match {
+      case Success(v) =>
+        LOG.info("shutting down the cluster....")
+        v.shutdown()
+      case Failure(v) =>
+        LOG.error("Could not start rest services", v)
+    }
+  }
+}
```
Comment (kkasravi) [inline services/src/test/scala/org/apache/gearpump/services/WorkersServiceSpec.scala]: Would be nice to provide an example of the JSON for each API

**Comment by clockfly** [comment]
build error?

**Comment by clockfly** [comment]
```
[INFO] [02/12/2015 07:04:39.576] [WorkerSpec] Actor system is started, 127.0.0.1, 44766
[ERROR] [02/12/2015 07:04:39.647] [WorkerSystem-akka.actor.default-dispatcher-4] [akka://WorkerSystem/user/$a] null
akka.actor.ActorInitializationException: exception during creation
	at akka.actor.ActorInitializationException$.apply(Actor.scala:164)
	at akka.actor.ActorCell.create(ActorCell.scala:596)
	at akka.actor.ActorCell.invokeAll$1(ActorCell.scala:456)
	at akka.actor.ActorCell.systemInvoke(ActorCell.scala:478)
	at akka.dispatch.Mailbox.processAllSystemMessages(Mailbox.scala:263)
	at akka.dispatch.Mailbox.run(Mailbox.scala:219)
	at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:393)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at akka.util.Reflect$.instantiate(Reflect.scala:66)
	at akka.actor.ArgsReflectConstructor.produce(Props.scala:352)
	at akka.actor.Props.newActor(Props.scala:252)
	at akka.actor.ActorCell.newActor(ActorCell.scala:552)
	at akka.actor.ActorCell.create(ActorCell.scala:578)
	... 9 more
Caused by: java.util.NoSuchElementException: None.get
	at scala.None$.get(Option.scala:347)
	at scala.None$.get(Option.scala:345)
	at org.apache.gearpump.cluster.worker.Worker.<init>(Worker.scala:56)
	... 18 more
```

**Comment by clockfly** [comment]
Merge for now, and fix this later.

**Pull request merged**